### PR TITLE
lint ! and != in if expressions with else branches

### LINKED ISSUE #469: if x != y { a }  else { b } should be if x == y { b } else { a }
Suggest using the equality comparison operator instead of inequality if the arms can be trivially swapped. Inequality is one more indirection to think about, so using equality is easier to comprehend.

### PR DESCRIPTION
**Pull request by oli-obk**
fixes #469

### CODE REVIEW

File: src/swap_if_arms.rs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+use rustc::lint::*;
+use syntax::attr::*;
+use syntax::ast::*;
+
+use utils::span_help_and_lint;
+
+/// **What it does:** Warns on enum variants that are prefixed or suffixed by the same characters
+///
+/// **Why is this bad?** Enum variant names should specify their variant, not the enum, too.
```
Comment (mcarton) [inline src/swap_if_arms.rs]: Copy and paste :smile:

File: src/swap_if_arms.rs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+impl EarlyLintPass for SwapIfArms {
+    fn check_expr(&mut self, cx: &EarlyContext, item: &Expr) {
+        if let ExprKind::If(ref cond, _, Some(ref els)) = item.node {
+            if let ExprKind::Block(..) = els.node {
+                match cond.node {
+                    ExprKind::Unary(UnOp::Not, _) => {
+                        span_help_and_lint(cx,
+                                           SWAP_IF_ARMS,
+                                           item.span,
+                                           "Unnecessary boolean Not operation",
```
Comment (Manishearth) [inline src/swap_if_arms.rs]: lowercase n?

File: src/swap_if_arms.rs [hunk 3]
```diff
@@ -0,0 +1,53 @@
+//! lint on if branches that could be swapped so no `!` operation is necessary on the condition
+
+use rustc::lint::*;
+use syntax::attr::*;
+use syntax::ast::*;
+
+use utils::span_help_and_lint;
+
+/// **What it does:** Warns on enum variants that are prefixed or suffixed by the same characters
```
Comment (Manishearth) [inline src/swap_if_arms.rs]: docs need updating

File: src/swap_if_arms.rs [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+///
+/// **Why is this bad?** Enum variant names should specify their variant, not the enum, too.
+///
+/// **Known problems:** None
+///
+/// **Example:** enum Cake { BlackForestCake, HummingbirdCake }
+declare_lint! {
+    pub SWAP_IF_ARMS, Warn,
```
Comment (Manishearth) [inline src/swap_if_arms.rs]: perhaps something more like IF_NOT_ELSE? Our lint names usually describe the cause, not the solution.

File: src/swap_if_arms.rs (hunk 1 above)
Comment (oli-obk) [inline src/swap_if_arms.rs]: whoops

**Comment by oli-obk** [comment]
addressed comments

**Pull request merged**